test/e2e/here-maps.e2e.test.js

```diff
@@ -0,0 +1,190 @@
+const { test, expect } = require('@playwright/test');
+
+test.describe('HERE Maps API Integration', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('/api/here-maps');
+    await page.waitForLoadState('networkidle');
+  });
+
+  test('should load HERE Maps page and display basic elements', async ({ page }) => {
+    // Basic page checks
+    await expect(page).toHaveTitle(/Here Maps/);
+    await expect(page.locator('h2')).toContainText('HERE Maps API');
+
+    // Check for Font Awesome icon
+    const mapIcon = page.locator('h2 i.fas.fa-map-marked');
+    await expect(mapIcon).toBeVisible();
+
+    // Check for navigation buttons
+    const developerPortalBtn = page.locator('a[href*="developer.here.com"]').first();
+    const imageMapParamsBtn = page.locator('a[href*="developer.here.com/documentation"]');
+    await expect(developerPortalBtn).toBeVisible();
+    await expect(developerPortalBtn).toContainText('HERE Developer Portal');
+    await expect(imageMapParamsBtn).toBeVisible();
+    await expect(imageMapParamsBtn).toContainText('Image Map Parameters');
+
+    // Check for subtitle
+    await expect(page.locator('h3')).toContainText('Map using Here Interactive Map Service');
+
+    // Check for map container
+    const mapContainer = page.locator('#map');
+    await expect(mapContainer).toBeVisible();
+
+    // Verify map container has dimensions (style attribute contains width and height)
+    const mapStyle = await mapContainer.getAttribute('style');
+    expect(mapStyle).toContain('width: 100vw');
+    expect(mapStyle).toContain('height: 50vh');
+
+    // Check for distance text
+    const distanceText = page.locator('text=Straight line distance between the Fremont Troll and Seattle Art Museum is');
+    await expect(distanceText).toBeVisible();
+
+    // Check for distance value element
+    const distanceElement = page.locator('#directLineDistance');
+    await expect(distanceElement).toBeVisible();
+  });
+
+  test('should load HERE Maps JavaScript API scripts', async ({ page }) => {
```

Comment:
Let's drop this, since it is for static content, and script upgrades shouldn't fail the tests if the functionality is the same.